lib/index.js

```diff
@@ -207,12 +236,22 @@ api.set = async ({
   } else {
     query['account.email'] = email;
   }
+
+  let tokenTypeValue;
+  if(type === 'nonce') {
+    tokens.push(token);
+    tokenTypeValue = tokens;
```

Comment:
I'm not sure if the `else` below is correct, since it looks like when we don't have any existing tokens we can have either an empty array or a missing property (the `null` case). So we probably need to tweak the else here with some smarter mongodb query magic.

```suggestion
    const oldValue = tokens.length > 0 ? tokens.slice() : null;
    query['meta.bedrock-authn-token.tokens.nonce'] = oldValue;
    tokens.push(token);
    tokenTypeValue = tokens;
```

Reply:
We might need to use `$eq` for the case where we have old tokens as well.

Reply:
@JSAssassin -- I think the above query should work, updated it to use `$in`.